client/state/reader/start/reducer.js

```diff
@@ -21,19 +24,30 @@ import { isValidStateWithSchema } from 'state/utils';
 /**
  * Tracks all known list objects, indexed by list ID.
  *
- * @param  {Object} state  Current state
+ * @param  {Array} state  Current state
  * @param  {Object} action Action payload
- * @return {Object}        Updated state
+ * @return {Array}        Updated state
  */
-export function items( state = {}, action ) {
+export function items( state = [], action ) {
 	switch ( action.type ) {
 		case READER_START_RECOMMENDATIONS_RECEIVE:
-			return Object.assign( {}, state, keyBy( action.recommendations, 'ID' ) );
+			// Filter out any recommendations we already have
+			const newRecommendations = filter( action.recommendations, ( incomingRecommendation ) => {
+				if ( findIndex( state, ( existingRecommendation ) => {
+					return existingRecommendation.ID === incomingRecommendation.ID;
+				} ) >= 0 ) {
+					return false;
+				}
+				return true;
+			} );
+
+			return state.concat( newRecommendations );
```

Comment:
might want to check to make sure newRecommendations has length > 0. that way we can avoid changing the state when no new recs come in.